src/Stratis.Bitcoin.Features.Dns/DnsSettings.cs

```diff
@@ -0,0 +1,104 @@
+﻿using System;
+using Microsoft.Extensions.Logging;
+using Stratis.Bitcoin.Configuration;
+
+namespace Stratis.Bitcoin.Features.Dns
+{
+    /// <summary>
+    /// Configuration related to the DNS feature.
+    /// </summary>
+    public class DnsSettings
+    {
+        /// <summary>The default value for the DNS listen port.</summary>
+        public const int DefaultDnsListenPort = 53;
+
+        /// <summary>The default value which a peer should have last have been connected before being blacklisted in DNS nodes.</summary>
+        public const int DefaultDnsPeerBlacklistThresholdInSeconds = 1800;
+
+        /// <summary>The value which a peer should have last have been connected before being blacklisted from the DNS nodes.</summary>
+        public int DnsPeerBlacklistThresholdInSeconds { get; set; }
+
+        /// <summary>Defines the host name for the node when running as a DNS Seed service.</summary>
+        public string DnsHostName { get; set; }
+
+        /// <summary><c>true</c> if the DNS Seed service should also run as a full node, otherwise <c>false</c>.</summary>
+        public bool DnsFullNode { get; set; }
+
+        /// <summary>Defines the port that the DNS server will listen on, by default this is 53.</summary>
+        public int DnsListenPort { get; set; }
+
+        /// <summary>Defines the nameserver host name used as the authoritative domain for the DNS seed service.</summary>
+        public string DnsNameServer { get; set; }
+
+        /// <summary>Defines the e-mail address used as the administrative point of contact for the domain.</summary>
+        public string DnsMailBox { get; set; }
+
+        /// <summary>The callback used to override/constrain/extend the settings provided by the Load method.</summary>
+        private Action<DnsSettings> callback = null;
+
+        /// <summary>
+        /// Constructs this object.
+        /// </summary>
+        public DnsSettings()
+        {
+        }
+
+        /// <summary>
+        /// Constructs this object whilst providing a callback to override/constrain/extend 
+        /// the settings provided by the Load method.
+        /// </summary>
+        /// <param name="callback">The callback used to override/constrain/extend the settings provided by the Load method.</param>
+        public DnsSettings(Action<DnsSettings> callback)
```

Comment:
Hi @quantumagi! Ok so I cloned your branch and saw that we use this constructor in:

`public static IFullNodeBuilder UseDns(this IFullNodeBuilder fullNodeBuilder, Action<DnsSettings> setup = null)`

But `UseDns()` **with** the callback is never actually used anywhere? Can we remove this perhaps? As to not introduce unused code into the code base.

Reply:
Removed.

Reply:
@quantumagi ah yes OK! I also saw it now in the ApiSettings PR.